### PR TITLE
Moved get/got geo code out of HandleImGui

### PR DESCRIPTION
**Pull request by DamianMadden**
Works for submodels in folders now

Resolves [AB#356](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/356)

### CODE REVIEW

File: src/vcState.h [hunk 1]
```diff
@@ -173,7 +174,7 @@ struct vcState
   vcDocks changeActiveDock;
 
   bool getGeo;
-  udGeoZone *pGotGeo;
+  vcModel *pGotGeo;
```
Comment (samsinsane) [inline src/vcState.h]: I'm not sure I understand why this was changed?

File: src/scene/vcFolder.cpp [hunk 2]
```diff
@@ -93,8 +93,24 @@ void vcFolder::AddToScene(vcState *pProgramState, vcRenderData *pRenderData)
...
+      vcModel *pModel = (vcModel *)pNode->pUserData;
+      if (pProgramState->pGotGeo == nullptr)
+      {
+        pProgramState->getGeo = false;
+        pProgramState->pGotGeo = pModel;
+        vcProject_UseProjectionFromItem(pProgramState, pModel);
+      }
+    }
+
+    if (pProgramState->pGotGeo != nullptr)
```
Comment (samsinsane) [inline src/scene/vcFolder.cpp]: Why is this after the above block now? It used to be before?

File: src/vcState.h (hunk 1 above)
Comment (OsiJr) [reply src/vcState.h]: Changing this type is wrong. Lots of other things are sources for this and I already have changes that change this to support more types.

File: src/scene/vcFolder.cpp (hunk 2 above)
Comment (DamianMadden) [reply src/scene/vcFolder.cpp]: I was mistakenly thinking we needed to changeprojection on the source model, which previously might have happened after cycling through every scene item and finding the source model again before unsetting gotGeo

File: src/scene/vcFolder.cpp [hunk 3]
```diff
@@ -93,8 +93,25 @@ void vcFolder::AddToScene(vcState *pProgramState, vcRenderData *pRenderData)
 
     HandleNodeSelection(pProgramState, pNode);
 
+    if (pProgramState->pGotGeo != nullptr)
+    {
+      ((vcSceneItem *)(pNode->pUserData))->ChangeProjection(*pProgramState->pGotGeo);
+    }
+    else if (pProgramState->pGotGeo == nullptr && pProgramState->getGeo && pNode->itemtype == vdkPNT_PointCloud && ((vcModel *)(pNode->pUserData))->m_pPreferredProjection != nullptr)
```
Comment (samsinsane) [inline src/scene/vcFolder.cpp]: The `pProgramState->pGotGeo == nullptr` is unnecessary due to the `if` statement.

**Pull request merged**